ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/events/EventsRange.scala

```diff
@@ -69,18 +71,23 @@ private[events] object EventsRange {
       range: EventsRange[Long],
       pageSize: Int,
   ): Connection => Vector[A] = connection => {
-    val minPageSize = 10 min pageSize max (pageSize / 10)
-    val guessedPageEnd = range.endInclusive min (range.startExclusive + pageSize)
-    val arithPage =
-      read(range copy (endInclusive = guessedPageEnd), None, Some(pageSize))(connection)
-    val found = arithPage.size
-    if (guessedPageEnd == range.endInclusive || found >= minPageSize)
-      arithPage
-    else
-      arithPage ++ read(
-        range copy (startExclusive = guessedPageEnd),
-        Some(minPageSize - found),
-        Some(minPageSize - found),
-      )(connection)
+
```

Comment:
this changes completely how we interpret semantically the coming in range.
previously: try to grab a reasonable bite from the range, with first a limitless but fetchhinted range query from start to start+pageSize, to see whether we have enough results in that range, then if not do an additional query which is limited, to bite a bigger one with a limitted query, so that alltogether we have pageSize items, or the range is exhausted.
Note: this algorithm has no intention to exhaust the range.
now: 
do fetch hinted limitless queries, until we exhaust the range

Is this intentional? Isn't that dangerous to fetch here everything?

I did not understand the relevance of the previous approach regarding these various strategies regarding the specific combinations of limits/fetch hints, so it is probably a stupid question, but why is this approach with exclusively fetchinted limitless queries better?